doc/adrs/152_reactivate_tool_selection_dialog.adoc

```diff
@@ -0,0 +1,144 @@
+
+```ts
+if (tool.selectionDescriptionId) {
+  const showSelectionDialogEvent: ShowSelectionDialogEvent = {
+    type: 'SHOW_SELECTION_DIALOG',
+    activeTool: tool,
+  };
+  dispatch(showSelectionDialogEvent);
+} else {
+  invokeTool(tool, element.id, startingPosition);
+  diagramServer.actionDispatcher.dispatch({ kind: SOURCE_ELEMENT_ACTION });
+}
+```
+
+Then we open the selection dialog: 
+
+```ts
+if (selectionDialog === 'visible') {
+  selectModelElementDialog = (
+    <SelectionDialogWebSocketContainer
+      editingContextId={editingContextId}
+      selectionRepresentationId={(activeTool as CreateNodeTool).selectionDescriptionId}
+      targetObjectId={contextualPalette.element.targetObjectId}
+      onClose={() => {
+        dispatch({ type: 'CLOSE_SELECTION_DIALOG' } as CloseSelectionDialogEvent);
+      }}
+      onFinish={(selectedObjectId) => {
+        dispatch({
+          type: 'HANDLE_SELECTED_OBJECT_IN_SELECTION_DIALOG',
+          selectedObjectId,
+        } as HandleSelectedObjectInSelectionDialogEvent);
+      }}
+    />
+  );
+}
+```
+
+The tool was finaly invoked within this useEffect (depending on the selectedObjectId value):
+
+```ts
+  useEffect(() => {
+    if (selectedObjectId && activeTool && contextualPalette) {
+      invokeTool(activeTool, contextualPalette.element.id, contextualPalette.startingPosition);
+      diagramServer.actionDispatcher.dispatch({ kind: SOURCE_ELEMENT_ACTION });
+      const resetSelectedObjectInSelectionDialogEvent: ResetSelectedObjectInSelectionDialogEvent = {
+        type: 'RESET_SELECTED_OBJECT_IN_SELECTION_DIALOG',
+      };
+      dispatch(resetSelectedObjectInSelectionDialogEvent);
+    }
+  }, [activeTool, diagramServer, invokeTool, dispatch, selectedObjectId, contextualPalette]);
+```
+
+== Decision
+
+The new diagram architecture based on react-flow has changed significantly. The tools are now invoked in the _Palette_ component:
+
+```ts
+  const handleToolClick = (tool: GQLTool) => {
+    switch (tool.id) {
+      case 'edit':
+        onDirectEditClick();
+        break;
+      case 'semantic-delete':
+        showDeletionConfirmation(() => {
+          invokeDelete(diagramElementId, GQLDeletionPolicy.SEMANTIC);
+        });
+        break;
+      case 'graphical-delete':
+        invokeDelete(diagramElementId, GQLDeletionPolicy.GRAPHICAL);
+        break;
+      case 'expand':
+        collapseExpandElement(diagramElementId, GQLCollapsingState.EXPANDED);
+        break;
+      case 'collapse':
+        collapseExpandElement(diagramElementId, GQLCollapsingState.COLLAPSED);
+        break;
+      default:
+        invokeSingleClickTool(tool);
+        break;
+    }
+  };
+```
+
+In a similar way than the delete tool shows the deletion confirmation dialog, the palette should call a showSelectionDialog if the tool defines a selectionDescriptionID.
```

Comment:
In the react-flow integration, we will not have any reference to the term `selectionDialog` but instead to dialog. We can introduce a hook named `useDialog()` which should multiple properties:

- A `showDialog()` function to make the dialog visible
- A `hideDialog()` function to make it disappear
- A value `variables: GQLToolVariable[]` containing the variables to add to the backend

For that, we should modify the input `InvokeSingleClickOnDiagramElementToolInput` to remove the properties `selectedObjectId` and instead add `variables: [ToolVariable!]!` with the following type:

```
input ToolVariable {
  name: String!
  value: String!
  type: ToolVariableType!
}

enum ToolVariableType = STRING | OBJECT_ID | OBJECT_ID_ARRAY
```

Reply:
done